sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ReviewSqlQueriesForSecurityVulnerabilities.cs

```diff
@@ -85,11 +114,90 @@ protected override void Initialize(SonarAnalysisContext context)
                         c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, assignment.Left.GetLocation()));
                     }
                 }, SyntaxKind.SimpleAssignmentExpression);
+
+            context.RegisterCompilationStartAction(
+                cc =>
+                {
+                    if (!TryReadConfiguration(cc.Options))
+                    {
+                        return;
+                    }
+
+                    protobufFileIndex = 0;
+
+                    cc.RegisterSyntaxNodeActionInNonGenerated(
+                        c => WriteUCFG<BaseMethodDeclarationSyntax>(c, x => x.Body),
+                        SyntaxKind.ConstructorDeclaration,
+                        SyntaxKind.OperatorDeclaration);
+
+                    cc.RegisterSyntaxNodeActionInNonGenerated(
+                        c => WriteUCFG<MethodDeclarationSyntax>(c, x => (CSharpSyntaxNode)x.Body ?? x.ExpressionBody?.Expression),
+                        SyntaxKind.MethodDeclaration);
+
+                    cc.RegisterSyntaxNodeActionInNonGenerated(
+                        c => WriteUCFG<AccessorDeclarationSyntax>(c, node => node.Body),
+                        SyntaxKind.GetAccessorDeclaration,
+                        SyntaxKind.SetAccessorDeclaration);
+
+                    cc.RegisterSyntaxNodeActionInNonGenerated(
+                        c => WriteUCFG<PropertyDeclarationSyntax>(c, node => node.ExpressionBody?.Expression),
+                        SyntaxKind.PropertyDeclaration);
+                });
+        }
+
+        private void WriteUCFG<TDeclarationSyntax>(SyntaxNodeAnalysisContext context, Func<TDeclarationSyntax, CSharpSyntaxNode> getBody)
+            where TDeclarationSyntax : SyntaxNode
+        {
+            var declaration = (TDeclarationSyntax)context.Node;
+
+            var symbol = context.SemanticModel.GetDeclaredSymbol(declaration);
+
+            if (symbol == null ||
+                symbol.IsAbstract ||
+                symbol.IsExtern ||
+                !CSharpControlFlowGraph.TryGet(getBody(declaration), context.SemanticModel, out var cfg))
+            {
+                return;
+            }
+
+            var ucfg = new UCFG
```

Comment:
It would be neater if the _UCFGBuilder_ did all of the building; it looks a bit odd have this method create the _UCFG_ object then call the builder to do three different jobs.